### PR TITLE
notebook for subs

### PR DESCRIPTION
**Pull request by justicart**
Summary:

Test Plan:
---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/bolt-foundry/bolt-foundry/pull/908).
* __->__ #908
* #904
* #902
* #901
* #900

### CODE REVIEW

File: deno.jsonc [hunk 1]
```diff
@@ -117,8 +119,8 @@
     ]
   },
   "lock": {
-    "path": "./deno.lock"
-    "frozen": true
```
Comment (randallb) [inline deno.jsonc:121]: revert this change please.

Also, I found out that after you do `deno add` you just do `deno install --frozen=false`. This is how we'll prevent accidentally installing deps.

File: packages/bfDb/models/BfMediaNodeVideo.ts [hunk 2]
```diff
@@ -36,29 +38,36 @@ export class BfMediaNodeVideo<T = {}> extends BfNode<BfMediaNodeVideoProps<T>> {
...
+    const edge = await BfEdge.querySourceEdgesForNode(this);
+    return edge[0]?.props?.role?.toLowerCase();
+  }
+
+  private async getObjectKey(): Promise<string> {
+    const role = await this.getRole();
+    return `${this.metadata.bfGid}_${role}.mp4`;
+  }
+
+  private objectStorageClient = new Client();
```
Comment (randallb) [inline packages/bfDb/models/BfMediaNodeVideo.ts:51]: Kind of weird to have an object storage client per object... probably just want one in the module scope, or a static one.

File: packages/bfDb/models/BfMediaNodeVideo.ts [hunk 3]
```diff
@@ -36,29 +38,36 @@ export class BfMediaNodeVideo<T = {}> extends BfNode<BfMediaNodeVideoProps<T>> {
...
     logger.debug("getFilePath", filePath);
 
     return await Deno.stat(filePath).then(() => filePath).catch(() => null);
   }
 
-  getFile(): Promise<string> {
-    throw new BfError(
-      `Need to implement getFile() for ${this.constructor.name}`,
-    );
+  async getFile(): Promise<string> {
```
Comment (randallb) [inline packages/bfDb/models/BfMediaNodeVideo.ts:60]: you could change this to be `doesFileExistInObjectStorage` and then return true / false

File: experimental/justin/touch-i-remember-touch.ipynb
Comment (randallb) [inline experimental/justin/touch-i-remember-touch.ipynb:1]: Probably don't want to check in the specific account ID, though it doesn't really matter.

**Review by randallb — APPROVED** [review]

File: packages/bfDb/models/BfMediaNodeVideo.ts (hunk 3 above)
Comment (justicart) [reply packages/bfDb/models/BfMediaNodeVideo.ts:60]: reverted this because I wasn't using it this way :(
